pkg/operator/awspodidentity/controller.go

```diff
@@ -0,0 +1,297 @@
+	}
+
+	p := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return isManaged(e.MetaNew)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return isManaged(e.Meta)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return isManaged(e.Meta)
+		},
+	}
+
+	// Create a namespace local cache separate from the Manager cache
+	// A namespace scoped cache can still handle cluster scoped resources
+	cache, err := cache.New(config, cache.Options{Namespace: operatorNamespace})
+	if err != nil {
+		return err
+	}
+	allFiles := append(staticFiles, templateFiles...)
+	for _, file := range allFiles {
+		objBytes := v410_00_assets.MustAsset(file)
+		obj, _, err := defaultCodec.Decode(objBytes, nil, nil)
+		if err != nil {
+			return err
+		}
+
+		informer, err := cache.GetInformer(obj)
+		if err != nil {
+			return err
+		}
+
+		err = c.Watch(&source.Informer{Informer: informer}, &handler.EnqueueRequestForObject{}, p)
+		if err != nil {
+			return err
+		}
+	}
+
+	mgr.Add(&awsPodIdentityController{reconciler: r, cache: cache, logger: logger})
+
+	return nil
+}
+
+func isManaged(meta metav1.Object) bool {
+	// all managed resources are named pod-identity-webhook
+	if meta.GetName() == "pod-identity-webhook" {
+		return true
+	}
+	return false
+}
+
+type staticResourceReconciler struct {
+	clientset            *kubernetes.Clientset
+	logger               log.FieldLogger
+	eventRecorder        events.Recorder
+	deploymentGeneration int64
+	imagePullSpec        string
+}
+
+var _ reconcile.Reconciler = &staticResourceReconciler{}
+
+func (r *staticResourceReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	r.logger.Debugf("reconciling after watch event %#v", request)
+	err := r.ReconcileResources()
+	if err != nil {
+		r.logger.Errorf("reconciliation failed, retrying in %s", retryInterval.String())
+		return reconcile.Result{RequeueAfter: retryInterval}, err
+	}
+	return reconcile.Result{}, nil
+}
+
+func (r *staticResourceReconciler) ReconcileResources() error {
+	applyResults := resourceapply.ApplyDirectly(
+		(&resourceapply.ClientHolder{}).WithKubernetes(r.clientset),
+		r.eventRecorder,
+		v410_00_assets.Asset,
+		staticFiles...,
+	)
+	for _, result := range applyResults {
+		if result.Error != nil {
+			r.logger.WithError(result.Error).Errorf("error reconciling %s", result.Type)
+			return result.Error
+		}
+		if !result.Changed {
+			continue
+		}
+		r.logger.Infof("%s reconciled successfully", result.Type)
+	}
+
+	// "v4.1.0/aws-pod-identity-webhook/deployment.yaml"
+	requestedDeployment := resourceread.ReadDeploymentV1OrDie(v410_00_assets.MustAsset("v4.1.0/aws-pod-identity-webhook/deployment.yaml"))
+	requestedDeployment.Spec.Template.Spec.Containers[0].Image = r.imagePullSpec
+	resultDeployment, modified, err := resourceapply.ApplyDeployment(r.clientset.AppsV1(), r.eventRecorder, requestedDeployment, r.deploymentGeneration, false)
+	r.deploymentGeneration = resultDeployment.Generation
```

Comment:
Should we read and store the Generation field before checking whether an error ocorred?